juju/js-libjuju
# update dependencies

## Description

- Update dependencies: `@canonical/macaroon-bakery` and `typedoc` (used for the documentation website)

### QA

Checkout this PR
- In the root of the PR codebase: `npm install && npm run build && yarn link`
- Install a beta 3.0 of Juju: `snap install juju --channel=3.0/beta`
- Make sure that your Juju instance is working: `juju dashboard`
- Clone the jaas-dasbhoard: `git clone https://github.com/canonical/jaas-dashboard && cd jaas-dashboard`
- Add a new file at `public/config.local.js`:
```js
var jujuDashboardConfig = {
  controllerAPIEndpoint: "wss://<your-dashboard-host:port>/api",
  baseAppURL: "/",
  identityProviderAvailable: false,
  identityProviderURL: "",
  isJuju: true,
};
```
- Replace the lines `4-11` of `juju/index.js` to:
```js
import actions from "@canonical/jujulib/dist/api/facades/action-v7";
import allWatcher from "@canonical/jujulib/dist/api/facades/all-watcher-v2";
import annotations from "@canonical/jujulib/dist/api/facades/annotations-v2";
import applications from "@canonical/jujulib/dist/api/facades/application-v14";
import client from "@canonical/jujulib/dist/api/facades/client-v5";
import cloud from "@canonical/jujulib/dist/api/facades/cloud-v7";
import controller from "@canonical/jujulib/dist/api/facades/controller-v11";
import modelManager from "@canonical/jujulib/dist/api/facades/model-manager-v9";
```
- In the root of `jaas-dashboard` run: `yarn && yarn link @canonical/jujulib`
- Run the dasbhoard: `yarn start`
- Go to localhost:3000
- Make sure that the dashboard is working without errors

Fixes #80
